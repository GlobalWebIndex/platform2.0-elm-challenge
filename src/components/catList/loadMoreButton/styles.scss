@import '../../../assets/stylesheets/classes.scss';

.loadMore {
  @extend %baseButton;

  background-color: #28a745;
  border: 1px solid #28a745;
  color: #fff;

  &:disabled {
    box-shadow: none;
    opacity: 0.65;
    pointer-events: none;
  }

  &:focus:not(:disabled) {
    outline: 0;
    box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
  }

  &:hover:not(:disabled) {
    background-color: #218838;
    border-color: #1e7e34;
    color: #fff;
  }
}
