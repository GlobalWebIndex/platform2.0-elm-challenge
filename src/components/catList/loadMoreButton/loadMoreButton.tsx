import React from 'react';
import styles from './styles.scss';

type LoadMoreButtonProps = {
  isDisabled: boolean;
  onLoadMore: () => void;
};

export const LoadMoreButton = ({isDisabled, onLoadMore}: LoadMoreButtonProps): JSX.Element => (
  <button className={styles.loadMore} disabled={isDisabled} onClick={onLoadMore}>
    Load more
  </button>
);
