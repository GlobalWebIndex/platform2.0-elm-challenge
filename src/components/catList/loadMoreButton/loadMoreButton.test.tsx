import React from 'react';
import {render, screen} from '@testing-library/react';
import {userEvent} from '@testing-library/user-event';
import {LoadMoreButton} from './loadMoreButton';

describe('<LoadMoreButton />', () => {
  const props = {isDisabled: false, onLoadMore: jest.fn()};

  it('should render the button', () => {
    const {container} = render(<LoadMoreButton {...props} />);

    expect(container).toMatchSnapshot();
  });

  describe('when passed `isDisabled`', () => {
    const propsDisabled = {...props, isDisabled: true};

    it('should render the button as "disabled"', () => {
      render(<LoadMoreButton {...propsDisabled} />);

      expect(screen.getByText('Load more')).toBeDisabled();
    });

    describe('when the button is clicked', () => {
      it('should not fire `onLoadMore` from props', async () => {
        render(<LoadMoreButton {...propsDisabled} />);

        await userEvent.click(screen.getByText('Load more'));

        expect(propsDisabled.onLoadMore).not.toHaveBeenCalled();
      });
    });
  });

  describe('when the button is clicked', () => {
    it('should fire `onLoadMore` from props', async () => {
      render(<LoadMoreButton {...props} />);

      await userEvent.click(screen.getByText('Load more'));

      expect(props.onLoadMore).toHaveBeenCalledTimes(1);
      expect(props.onLoadMore).toHaveBeenCalledWith(expect.any(Object));
    });
  });
});
