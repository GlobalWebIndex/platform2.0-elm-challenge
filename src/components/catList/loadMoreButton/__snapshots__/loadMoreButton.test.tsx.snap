// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMoreButton /> should render the button 1`] = `
<div>
  <button
    class="loadMore"
  >
    Load more
  </button>
</div>
`;
